src/Liquidity sawp Test.sol

```diff
@@ -0,0 +1,177 @@
+    }
+
+  
+
+
+    function testMintFunction() public {
+        // Prank as user1 to mint tokens to user2
+        vm.prank(user1);
+        token0.mint(user2, SWAP_AMOUNT_IN);
+        // Check token balances after minting
+        uint256 token0BalanceUser2 = token0.balanceOf(user2);
+
+        assertEq(token0BalanceUser2, SWAP_AMOUNT_IN); // user2 should have 500 TK0
+    }
+
+    function testSwapFunction() public {
+        // Ensure user1 has sufficient token0 balance for swap
+        uint256 initialToken0BalanceUser1 = token0.balanceOf(user1);
+        if (initialToken0BalanceUser1 < SWAP_AMOUNT_IN) {
+            uint256 mintAmount = SWAP_AMOUNT_IN - initialToken0BalanceUser1;
+            token0.mint(user1, mintAmount);
+        }
+
+        // Prank as user1 to perform a swap
+        address to = user1;
+
+        uint256 initialToken1BalanceUser1 = token1.balanceOf(user1);
+
+        vm.prank(user1);
+        liquidityPool.swap(POOL_NAME, address(token0), address(token1), SWAP_AMOUNT_IN, SWAP_AMOUNT_OUT_MIN, to);
+
+        // Check token balances after swap
+        uint256 finalToken0BalanceUser1 = token0.balanceOf(user1);
+        uint256 finalToken1BalanceUser1 = token1.balanceOf(user1);
+
+        assertEq(finalToken0BalanceUser1, initialToken0BalanceUser1 - SWAP_AMOUNT_IN); // Expect 500 TK0 to be swapped out
+        assertTrue(finalToken1BalanceUser1 > initialToken1BalanceUser1); // Adjust based on the amountOut calculated
+    }
+
+    // Additional Tests
+
+    function testMultiHopSwapPossible() public {
+        bool isPossible = liquidityPool.isMultiHopSwapPossible(POOL_NAME);
+        // Assuming the mock implementation always returns false
+        assertTrue(!isPossible);
+    }
+
+    function testMintLPRewards() public {
+        // Setup and perform a swap as in testSwapFunction
+
+        uint256 initialToken0BalanceUser1 = token0.balanceOf(user1);
+        if (initialToken0BalanceUser1 < SWAP_AMOUNT_IN) {
+            uint256 mintAmount = SWAP_AMOUNT_IN - initialToken0BalanceUser1;
+            token0.mint(user1, mintAmount);
+        }
+
+        address to = user1;
+        vm.prank(user1);
+        liquidityPool.swap(POOL_NAME, address(token0), address(token1), SWAP_AMOUNT_IN, SWAP_AMOUNT_OUT_MIN, to);
+
+        // Verify LP rewards minted
+        uint256 lpRewardsBalance = lpRewardsToken.balanceOf(user1);
+        assertTrue(lpRewardsBalance > 0);
+    }
+
+    function testFuzzSwap(uint256 amountIn) public {
+        vm.assume(amountIn > 0 && amountIn < TOKEN_AMOUNT);
+
+        address to = user1;
+        vm.prank(user1);
+        liquidityPool.swap(POOL_NAME, address(token0), address(token1), amountIn, SWAP_AMOUNT_OUT_MIN, to);
+
+        uint256 finalToken0BalanceUser1 = token0.balanceOf(user1);
+        uint256 finalToken1BalanceUser1 = token1.balanceOf(user1);
+
+        assertTrue(finalToken1BalanceUser1 > 0);
+        assertTrue(finalToken0BalanceUser1 <= TOKEN_AMOUNT - amountIn);
+    }
+
```

Comment:
Use kirubels swap contract for calling swap and checking for failure on a swap request which goes out of range